src/vision/shiny_future.md

```diff
@@ -0,0 +1,66 @@
+
+["guessing game"]: https://doc.rust-lang.org/book/ch02-00-guessing-game-tutorial.html
+
+**Key observations:**
+
+* Async is integrated into standard Rust learning materials like RBE, not its own silo.
+* Standard ways to stand up a runtime and web server, no choices to make (at first...)
+    * *FIXME:* We should cover how she is able to easily change to other runtimes later and things keep working
+
+## Sally learns about spawn
+
+Still playing around, Sally starts to extend her 'guessing game' project with some new functionality. She adds a function that launches a few web requests to help her pick the random number and get a nice "quote of the day" to include in the website:
+
+```rust,ignore
+async fn guessing_game_configuration() -> std::io::Result<(usize, String)> {
+    let random_number = async_io::request::fetch("https://random-number.org/").await?;
+    let quotd = async_io::request::fetch("https://qotd.org/").await?;
+    Ok((random_number, qotd))
+}
+```
+
+Here, she is using the standard API for launching web requests. When she puts this into her IDE, she gets a lint warning:
+
+```notrust
+warning[E2222]: synchronized request fetch
+ --> src/main.rs:2:13
+  |
+2 |     let random_number = async_io::request::fetch("https://random-number.org/").await?;
+  |                                                                                ------
+3 |     let quotd = async_io::request::fetch("https://qotd.org/").await?;
+  |                                                               ------
+  |
+  = help: these two request fetches will be synchronized, which probably isn't what you want
+2 |     let random_number = async_io::request::fetch("https://random-number.org/").spawn().await?;
+  |                                                                                ----- consider introducing calls to `spawn`
+3 |     let quotd = async_io::spawn(async_io::request::fetch("https://qotd.org/")).spawn().await?;
+  |                                                                                ----- here too
+```
+
+Reading a bit more into the background for E2222, she learns about how async-await in Rust works differently from async-await in javaScript. She learns about spawning. etc etc.
+
+**Key observations:**
+
+* Having standard or recommended APIs will let us give better suggestions.
+* Exposing things like `spawn` as methods feels right to nikomatsakis =) 
+
+## Sally goes deeper with the Rust book
+
+Sally wants to get past simple examples, so she opens up the Rust book. She learns more about modules and ownership, while flipping over to the "Asynchronous Programming" section from time to time to learn more about that. 
+
+**Key observations:**
+
+* *FIXME:* This is really just wishful thinking. How do we actually make this happen? Is going to the book even realistic? Maybe this would be a good place to talk about helpful error messages, thoughtful suggetions from the compiler ('did you want to spawn?')
```

Comment:
I'd assume that some amount of people will use the Rust book as a primary introductory learning resource. Other people won't touch a book no matter how good it is. FWIW I think it's realistic to integrate the async book with the Rust Book, in the future.